Revise: reorder tabs and tab stops in profile preferences

SlySven opened the pull request:
Switch some uses of QGridLayout <==> QVBoxLayout depending on whether there is more or only one column of items.

Moved the "Special options" tab to the last position.

Switch some single columns of items to become two columns so that the overall vertical size needed for the dialogue can be reduced - by around a quarter.

Moved the Discord and IRC options to a new tab "Chat".

**The last two are important as they allow the overall dialogue to be smaller and thus may enable us to close: #3467, #3304.**

Removed unused groupBox that has something about deleting map backups. It is not handled or mentioned in the current code at all and just sits there as a disabled widget in the mapper tab.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

For discussion and comparison here are the tab-order screen shots of the existing and the proposed layouts:

Tab Index 0 - "General"
![Screenshot_exisiting_preferences_tab_1](https://user-images.githubusercontent.com/6163092/85360961-87c38d00-b512-11ea-96f3-8d9f30e638f9.png) ![Screenshot_proposed_preferences_tab_1](https://user-images.githubusercontent.com/6163092/85360976-8abe7d80-b512-11ea-9565-3b4c79d73056.png)

Tab Index 1 - "Input line"
![Screenshot_exisiting_preferences_tab_2](https://user-images.githubusercontent.com/6163092/85360959-872af680-b512-11ea-944c-3ca75ec217c6.png) ![Screenshot_proposed_preferences_tab_2](https://user-images.githubusercontent.com/6163092/85360975-8a25e700-b512-11ea-9dcb-6fb675d90263.png)

Tab Index 2 - "Main display"
![Screenshot_exisiting_preferences_tab_3](https://user-images.githubusercontent.com/6163092/85360956-872af680-b512-11ea-8d82-a1a4453a66b8.png) ![Screenshot_proposed_preferences_tab_3](https://user-images.githubusercontent.com/6163092/85360974-8a25e700-b512-11ea-973a-a52812fc6328.png)

Tab Index 3 - "Editor"
![Screenshot_exisiting_preferences_tab_4](https://user-images.githubusercontent.com/6163092/85360955-86926000-b512-11ea-829b-6a8979bd09ce.png) ![Screenshot_proposed_preferences_tab_4](https://user-images.githubusercontent.com/6163092/85360973-898d5080-b512-11ea-99ae-e79842507ddc.png)

Tab Index 4 - "Color view"
![Screenshot_exisiting_preferences_tab_5](https://user-images.githubusercontent.com/6163092/85360954-86926000-b512-11ea-8355-36f922c0d771.png) ![Screenshot_proposed_preferences_tab_5](https://user-images.githubusercontent.com/6163092/85360971-898d5080-b512-11ea-9798-ac44647600e9.png)

Tab Index 5 - "Mapper"
![Screenshot_exisiting_preferences_tab_6](https://user-images.githubusercontent.com/6163092/85360952-86926000-b512-11ea-9be9-1f2db0de5582.png) ![Screenshot_proposed_preferences_tab_6](https://user-images.githubusercontent.com/6163092/85360969-88f4ba00-b512-11ea-88d6-c7e01fd7cf25.png)

Tab Index 6 - "Mapper colors"
![Screenshot_exisiting_preferences_tab_7](https://user-images.githubusercontent.com/6163092/85360950-85f9c980-b512-11ea-8a93-5b610f10f3f5.png) ![Screenshot_proposed_preferences_tab_7](https://user-images.githubusercontent.com/6163092/85360967-88f4ba00-b512-11ea-95a6-ac94152c2b06.png)

NEW - Tab Index 7 "Chat"
![Screenshot_proposed_preferences_tab_8](https://user-images.githubusercontent.com/6163092/85360966-885c2380-b512-11ea-9585-c46a345739d2.png)

MOVED Tab Index (To be 8 - is 8 currently) "Connection"
![Screenshot_exisiting_preferences_tab_9](https://user-images.githubusercontent.com/6163092/85360947-84c89c80-b512-11ea-834d-4459d654e449.png) ![Screenshot_proposed_preferences_tab_9](https://user-images.githubusercontent.com/6163092/85360964-885c2380-b512-11ea-9cef-126898748278.png)

MOVED Tab Index (To be 9 - is 7 currently) "Special options"
![Screenshot_exisiting_preferences_tab_8](https://user-images.githubusercontent.com/6163092/85360948-85f9c980-b512-11ea-9e9a-5918e9890838.png) ![Screenshot_proposed_preferences_tab_10](https://user-images.githubusercontent.com/6163092/85360963-87c38d00-b512-11ea-8935-d3656a3cedc8.png)

SlySven committed:
Revise: reorder tabs and tab stops in profile preferences

Switch some uses of QGridLayout <==> QVBoxLayout depending on whether there
is more or only one column of items.

Moved the "Special options" tab to the last position.

Switch some single columns of items to become two columns so that the
overall vertical size needed for the dialogue can be reduced.

Moved the Discord and IRC options to a new tab "Chat".

These last two are important as they allow the overall dialogue to be
smaller and thus may enable us to close: #3467, #3304.

Removed unused groupBox that has something about deleting map backups. It
is not handled or mentioned in the current code at all and just sits there
as a disabled widget in the mapper tab.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

vadi2 commented:
Some early feedback before I have a chance to do the full review - the dialog is pretty stretched out now and feels too wide:

![Profile preferences - Achaea_176](https://user-images.githubusercontent.com/110988/85422244-ad35b280-b575-11ea-97f6-3a15f2337a92.png)
![Profile preferences - Achaea_177](https://user-images.githubusercontent.com/110988/85422257-b0c93980-b575-11ea-83f0-b656d2a30084.png)

Xabre666 commented:
I don't think it's too wide, however preferences window fitting an HD screen (1366x768) instead of being cut off is a very welcome change for users with such screen resolution.

SlySven commented:
> The dialog is pretty stretched out now and feels too wide:

How wide does it appear as for you? {On KDE desktop you can get it to show the size and location of a widget as it is moved/resized} If you try to resize it to be narrower can you *get it to go smaller*? Switching through the tabs and trying to do that may give a hint as to the widgets that are forcing the dialogue to a certain width (as they fill the space allocated for them and block further shrinkage) and if we can identify them we might be able to tweak things a bit more...

Potential tight spots might be (numbers relate to proposed layout):
* Checkboxes **9** and **12** on the "General" tab
* Placeholder text for **28** on the "Input line" tab
* The label associated with **145** on the "Special options" tab

vadi2 commented:
Yes, it's not possible to make it smaller - wouldn't have said anything if it was 😉

vadi2 commented:
@Kebap could you review this?

Xabre666 commented:
Pease, please, please, approve an merge this before the freeze.

vadi2 commented:
It still has outstanding work it to be done - the window grew wider, not thinner. Help in this area would be welcome to get it in before the merge window closes.

SlySven committed:
Revise: try to make the preferences dialogue narrower

Break up a couple of texts into more than one line.

Set the initial size to be the smallest I can get in use on my Linux
Desktop - others' mileage may vary!

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven commented:
:hammer_and_wrench: I am just splitting the potentially troublesome lines of text I have identified down into two lines - and left an advisory note for translators that they too have to do this. However the limiting factor for shrinking things within the Qt Designer utility now is the three massively oversized icons in the "Connection" tab - they do get resized down, I think,  by this code fragment at the top of the `dlgProfilePreferences` constructor (the same is also done in the `dlgConnectionProfiles` and `dlgSystemMessageArea` constructors as well:
```cpp
{
    // init generated dialog
    setupUi(this);

    QPixmap holdPixmap;

    holdPixmap = *(this->notificationAreaIconLabelWarning->pixmap());
    holdPixmap.setDevicePixelRatio(5.3);
    this->notificationAreaIconLabelWarning->setPixmap(holdPixmap);

    holdPixmap = *(this->notificationAreaIconLabelError->pixmap());
    holdPixmap.setDevicePixelRatio(5.3);
    this->notificationAreaIconLabelError->setPixmap(holdPixmap);

    holdPixmap = *(this->notificationAreaIconLabelInformation->pixmap());
    holdPixmap.setDevicePixelRatio(5.3);
    this->notificationAreaIconLabelInformation->setPixmap(holdPixmap);
```

This seems to be something about high resolution icons - but it does make designing forms harder...

I seem to have got it down to be a minimum of 700x501 pixels but that is going to be dependent on a number of factors that are not entirely witihn controllable limits - for others on different OSes with different DEs, Qt preferences and locales I'd some variation...!

vadi2 commented:
> I am just splitting the potentially troublesome lines of text I have identified down into two lines 

If we take a step back... is this really the right approach? Or should we look to condense the text so it fits on one line instead?

SlySven commented:
Well the texts that I have *folded* in the Engineering English and the width of the form available for them are are now:
* Toolbar notification if Mudlet is
minimized and new data arrives
(half-width)

* Allow server to download and play media
(requires GMCP enabled below)
(half-width)

* This font is not monospace, which may not be ideal for playing some text games:
you can use it but there could be issues with aligning columns of text
(full-width)

* Show 'LUA OK' messages for Timers with
minimum interval of (h:mm:ss.zzz):
(half-width)

I do not see a lot of scope to shrink those down much - and even then that is only the Engineering English text - the translations for some locales will be wider (conversely some - Chinese and Taiwanese - should be much shorter!)

vadi2 commented:
@Mudlet/ui-review team will get on this case :sunglasses: 

Before:
> Toolbar notification if Mudlet is minimized and new data arrives

Condensed:

> Notify on new data

Tooltip: "Show a toolbar notification if Mudlet is minimized and new data arrives"

<br/><br/>Before:
> Allow server to download and play media (requires GMCP enabled below)

Condensed:

> Allow server to download and play media

Tooltip: "Allow server to download and play media, provided GMCP is enabled"

<br/><br/>Before:
> This font is not monospace, which may not be ideal for playing some text games: you can use it but there could be issues with aligning columns of text

It's a big warning, should stay that way, full width of the dialog.

<br/><br/>Before:
> Show 'LUA OK' messages for Timers with minimum interval of (h:mm:ss.zzz):

Condensed:

> Show debug messages for timers bigger than:

Tooltip: not sure on this one. 

@Kebap your take?

SlySven commented:
>> This font is not monospace, which may not be ideal for playing some text games: you can use it but there could be issues with aligning columns of text
>
> It's a big warning, should stay that way, full width of the dialog.

It still can fit the full width - the `QLabel` does span the whole dialogue (well the groupBox it is in less margins) but I put in a new line at the semicolon. Given that `QLabels` do not support wordwrapping (IIRC) the best compromise might be to centre the text horizontally. Though for consistency we ought to look to do the same for al the other temporary warning/informational messages that get shown in the same way...

> Before:
>> Show 'LUA OK' messages for Timers with minimum interval of (h:mm:ss.zzz):
>
> Condensed:
>> Show debug messages for timers bigger than:

Those two texts are not equivalent: the first says that the message will be shown for timers **>=** to the indicated time whereas the second says only show those **>** but not **=** to the time shown...

vadi2 commented:
> Those two texts are not equivalent: the first says that the message will be shown for timers >= to the indicated time whereas the second says only show those > but not = to the time shown...

Indeed, it's a tiny difference that is explained in the tooltip, the saved screen real-estate makes it worth it.

SlySven committed:
Merge branch 'development' into Revise_reorderTabsAndTabStopsInProfilePreferences

Conflicts resolved in:
* src/ui/profile_preferences.ui

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: tweek some texts in accordance with peer-review suggestions

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven commented:
@vadi2 have a look now - I have *largely* adopted your suggestions...

SlySven committed:
BugFix: fix a copy/paste error in XML file

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
BugFix: repair an error made when merging in development branch

And a typo when putting in a tool tip!

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

vadi2 reviewed (approved):
I think this looks great, thanks a lot for doing this and making the tab order sane!
